base/experimental.jl

```diff
@@ -52,30 +52,37 @@ macro aliasscope(body)
 end
 
 
-function sync_end(refs)
-    local c_ex
-    defined = false
-    t = current_task()
+function sync_end(refs::Vector{Any}, l::Threads.SpinLock)
     cond = Threads.Condition()
-    lock(cond)
-    nremaining = length(refs)
-    for r in refs
-        schedule(Task(()->begin
-            try
-                wait(r)
-                lock(cond)
-                nremaining -= 1
-                nremaining == 0 && notify(cond)
-                unlock(cond)
-            catch e
-                lock(cond)
-                notify(cond, e; error=true)
-                unlock(cond)
-            end
-        end))
+    while true
```

Comment:
This `while true` is conceptually invalid for this macro definition. This appears to possibly require changing `refs` into a proper Channel, rather than reimplementing it here with locks.

Reply:
What is the problem exactly?